cpu/fe310/irq_arch.c

```diff
@@ -108,14 +108,28 @@ void handle_trap(uint32_t mcause)
         }
     }
     else {
+        switch (mcause) {
+            case 8:  /* ECALL from user mode */
+            case 11: /* ECALL from machine mode */
```

Comment:
can we get defines for that :)

Reply:
Of course, the [risc-v encoding.h](https://github.com/riscv/riscv-opcodes/blob/master/encoding.h#L193-L207) doesn't define them, but I can extend that file with the common exceptions.

Reply:
Style nit: `case` labels and the `switch` statement should have the same indent level, only code after the `case` should have an indent.

Reply:
> Of course, the [risc-v encoding.h](https://github.com/riscv/riscv-opcodes/blob/master/encoding.h#L193-L207) doesn't define them, but I can extend that file with the common exceptions.

[Our own version](https://github.com/RIOT-OS/RIOT/blob/master/cpu/fe310/include/vendor/encoding.h#L863-L874) has them though…

Reply:
> Style nit: `case` labels and the `switch` statement should have the same indent level, only code after the `case` should have an indent.

That whole file could use a good round of uncrustify. Fixed it here for a start, I'll open an uncrustify PR later.